mllib/src/test/scala/org/apache/spark/ml/regression/RandomForestRegressorSuite.scala

```diff
@@ -105,6 +108,55 @@ class RandomForestRegressorSuite extends SparkFunSuite with MLlibTestSparkContex
       }
   }
 
+  test("Random Forest variance") {
```

Comment:
These tests will still pass if we ever implement a new impurity type, but the computed variance won't be correct. In fact, the tests in `DecisionTreeRegressorSuite` also fail to capture this. If `impurityStats.calculate()` doesn't always return the variance of the labels then the current implementation won't work. We should add a test here and in DT to ensure that the variance is computed correctly. The best way is probably to construct a fake tree, use some small fake data and ensure that the resulting variance is equal to the formula in the linked paper.